mecanum_drive_controller/src/mecanum_drive_controller.yaml

```diff
@@ -0,0 +1,121 @@
+    type: double,
+    default_value: 0.0,
+    description: "Timeout for controller references after which they will be reset. This is especially useful for controllers that can cause unwanted and dangerous behavior if reference is not reset, e.g., velocity controllers. If value is 0 the reference is reset after each run.",
+
+  }
+
+  joint_names: {
+    type: string_array,
+    default_value: [],
+    description: "Name of the wheels joints.",
+    read_only: false,
+
+  }
+
+  state_joint_names: {
+    type: string_array,
+    default_value: [],
+    description: "(optional) Specifies joints for reading states. This parameter is only relevant when state joints are different then command joint, i.e., when a following controller is used.",
+    read_only: true,
+
+  }
+
+  reference_names: {
+    type: string_array,
+    default_value: [],
+    description: " Names of the references, ex: high level vel commands from MoveIt, Nav2, etc.",
+    read_only: true,
+  }
+
+  interface_name: {
+    type: string,
+    default_value: "",
+    description: "Name of the interface used by the controller on joints and command_joints.",
+    read_only: true,
+  }
+
+  kinematics:
+    base_frame_offset:
+      x: {
+        type: double,
+        default_value: 0.0,
+        description: "Base frame offset along X axis of base_frame (base_link frame)." ,
+        read_only: true,
+      }
+      y: {
+        type: double,
+        default_value: 0.0,
+        description: "Base frame offset along Y axis of base_frame (base_link frame)." ,
+        read_only: true,
+      }
+      theta: {
+        type: double,
+        default_value: 0.0,
+        description: "Base frame offset along Theta axis of base_frame (base_link frame).",
+        read_only: true,
+      }
+
+    wheels_radius: {
+      type: double,
+      default_value: 0.0,
+      description: "Wheel's radius.",
+      read_only: false,
+
+    }
+
+    wheels_k: {
+      type: double,
+      default_value: 0.0,
+      description: "Wheels geometric param used in mecanum wheels' IK. lx and ly represent the distance from the robot's center to the wheels projected on
+      the x and y axis with origin at robots center respectively, wheels_k = lx+ly",
+      read_only: false,
+
+    }
+
+  base_frame_id: {
+    type: string,
+    default_value: "base_link",
+    description: "Base frame_id set to value of base_frame_id.",
+    read_only: false,
+
+  }
+  odom_frame_id: {
+    type: string,
+    default_value: "odom",
+    description: "Odometry frame_id set to value of odom_frame_id.",
+    read_only: false,
+
+  }
+  enable_odom_tf: {
+    type: bool,
+    default_value: true,
+    description: "Publishing to tf is enabled or disabled ?.",
+    read_only: false,
+
+  }
+
+  use_realigned_roller_joints: {
+    type: bool,
+    default_value: false,
+    description: "This mode is used when the mecanum wheels are simulated and we use realigned rollers to mimic mecanum wheels.",
```

Comment:
What? What is this thing doing exactly?

Reply:
at the moment there is no relevance for this variable as we are not setting Wheel Parameters from Urdf, hence will remove it.